app/services/credit_notes/refunds/stripe_service.rb

```diff
@@ -79,7 +79,21 @@ def should_process_refund?
       end
 
       def payment
-        @payment ||= credit_note.invoice.payments.order(created_at: :desc).first
+        return @payment if defined?(@payment)
+
+        @payment = if credit_note.invoice.payments.where(status: "succeeded").present?
+          credit_note.invoice.payments.where(status: "succeeded").order(created_at: :desc).first
```

Comment:
payment `status` attribute is the raw status from the provider and we should not rely on its content for our logic.... 

should we use `payable_payment_status` instead?

Reply:
Updated 👍